src/Sylius/Bundle/UiBundle/Resources/config/services/twig.xml

```diff
@@ -45,16 +52,19 @@
         <service id="Sylius\Bundle\UiBundle\Twig\LegacySonataBlockExtension">
             <argument>%sylius_ui.sonata_block.whitelisted_variables%</argument>
             <tag name="twig.extension" />
+            <deprecated package="sylius/ui-bundle" version="1.14">The service %service_id% is deprecated and will be removed in Sylius 2.0."</deprecated>
         </service>
 
         <service id="Sylius\Bundle\UiBundle\Twig\RedirectPathExtension">
-            <argument type="service" id="Sylius\Bundle\UiBundle\Storage\FilterStorage" />
+            <argument type="service" id="sylius.twig.storage.filter" />
             <argument type="service" id="router" />
             <tag name="twig.extension" />
         </service>
+        <service id="sylius.twig.extension.redirect_path" alias="Sylius\Bundle\UiBundle\Twig\RedirectPathExtension" />
 
         <service id="Sylius\Bundle\UiBundle\Storage\FilterStorage" public="false">
             <argument type="service" id="request_stack" />
         </service>
+        <service id="sylius.twig.storage.filter" alias="Sylius\Bundle\UiBundle\Storage\FilterStorage" />
```

Comment:
Check in 2.0 if we couldn't use the filter storage from GridBundle ([reference](https://github.com/Sylius/SyliusGridBundle/pull/342/files#diff-49ab24127ec11d9ffb12586e1c8ea7b4c10910ae0a60584e17ac344152dc630fR18)) and then remove it there and deprecate in 1.14.